yjit/src/codegen.rs

```diff
@@ -5198,28 +5191,30 @@ fn get_method_gen_fn()
 {
     // TODO: implement pattern matching for this
 
-    /*
-    // Specialization for C methods. See yjit_reg_method() for details.
-    yjit_reg_method(rb_cBasicObject, "!", jit_rb_obj_not);
-
-    yjit_reg_method(rb_cNilClass, "nil?", jit_rb_true);
-    yjit_reg_method(rb_mKernel, "nil?", jit_rb_false);
-
-    yjit_reg_method(rb_cBasicObject, "==", jit_rb_obj_equal);
-    yjit_reg_method(rb_cBasicObject, "equal?", jit_rb_obj_equal);
-    yjit_reg_method(rb_mKernel, "eql?", jit_rb_obj_equal);
-    yjit_reg_method(rb_cModule, "==", jit_rb_obj_equal);
-    yjit_reg_method(rb_cSymbol, "==", jit_rb_obj_equal);
-    yjit_reg_method(rb_cSymbol, "===", jit_rb_obj_equal);
-
-    // rb_str_to_s() methods in string.c
-    yjit_reg_method(rb_cString, "to_s", jit_rb_str_to_s);
-    yjit_reg_method(rb_cString, "to_str", jit_rb_str_to_s);
-    yjit_reg_method(rb_cString, "bytesize", jit_rb_str_bytesize);
-
-    // Thread.current
-    yjit_reg_method(rb_singleton_class(rb_cThread), "current", jit_thread_s_current);
-    */
+    // All these class constants are mutable statics.
+    unsafe {
+        // Specialization for C methods. See yjit_reg_method() for details.
+        //yjit_reg_method(rb_cBasicObject, "!", jit_rb_obj_not);
+
+        yjit_reg_method(rb_cNilClass, "nil?", jit_rb_true);
+        yjit_reg_method(rb_mKernel, "nil?", jit_rb_false);
+
+        yjit_reg_method(rb_cBasicObject, "==", jit_rb_obj_equal);
+        yjit_reg_method(rb_cBasicObject, "equal?", jit_rb_obj_equal);
+        yjit_reg_method(rb_mKernel, "eql?", jit_rb_obj_equal);
+        yjit_reg_method(rb_cModule, "==", jit_rb_obj_equal);
+        yjit_reg_method(rb_cSymbol, "==", jit_rb_obj_equal);
+        yjit_reg_method(rb_cSymbol, "===", jit_rb_obj_equal);
+        /*
+        // rb_str_to_s() methods in string.c
+        yjit_reg_method(rb_cString, "to_s", jit_rb_str_to_s);
+        yjit_reg_method(rb_cString, "to_str", jit_rb_str_to_s);
+        yjit_reg_method(rb_cString, "bytesize", jit_rb_str_bytesize);
+
+        // Thread.current
+        yjit_reg_method(rb_singleton_class(rb_cThread), "current", jit_thread_s_current);
+        */
+    }
```

Comment:
This doesn't seem like it needs to be wrapped in unsafe.

Reply:
Every class constant is a mutable static. If we don't wrap it in an unsafe block then every individual line needs to be wrapped in unsafe.